subprojects/workers/src/main/java/org/gradle/workers/internal/AdapterWorkerExecution.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.workers.internal;
+
+import org.gradle.api.tasks.WorkResult;
+import org.gradle.internal.reflect.Instantiator;
+import org.gradle.workers.WorkerExecution;
+
+import javax.inject.Inject;
+import java.util.concurrent.Callable;
+
+import static org.gradle.internal.classloader.ClassLoaderUtils.classFromContextLoader;
+
+public abstract class AdapterWorkerExecution implements WorkerExecution<AdapterWorkerParameters>, ProvidesWorkResult {
```

Comment:
could you add some class docs about what this does/is for?  We're running out of good descriptive names

Reply:
Done.